### PR TITLE
bfsbdump: add ECID dump

### PR DESCRIPTION
**Pull request by lsun100**
THis commit adds ECID dump which is supported in BlueField-3.

### CODE REVIEW

File: bfsbdump [hunk 1]
```diff
@@ -53,6 +53,15 @@ def GetPlatformName():
...
+    pos1 = str.find('656369643031000e')
+    pos2 = str.find('656369643233000e')
+    if (pos1 >= 0) and (pos2 >= 0):
+        return (str[pos1+16:pos1+32]) + str[pos2+16:pos2+32]
+    else:
+        return ""
+
 class ChipSecureBootStatus:
 
     def __init__(self):
```
Comment (kblaiech) [inline bfsbdump:67]: I suggest we add the ECID as an Object attribute.

`
def __init__(self):
        self.PlatformName = GetPlatformName()
        self.ecid = GetEcid()
`

File: bfsbdump [hunk 2]
```diff
@@ -134,6 +143,9 @@ class ChipSecureBootStatus:
                 a = (self.DotFsblVersion >>  0) & 0xFF, b = (self.DotFsblVersion >>  8) & 0xFF,
                 c = (self.DotFsblVersion >> 16) & 0xFF, d = (self.DotFsblVersion >> 24) & 0xFF ))
         print('DOT ID               : ' + str(binascii.hexlify(self.DotChipId).decode('ascii')))
+        ecid = GetEcid()
+        if ecid:
+            print('ECID                 : ' + ecid)
```
Comment (kblaiech) [inline bfsbdump:148]:
```
if self.ecid:
            print('ECID                 : ' + self.ecid)
```

File: bfsbdump [hunk 3]
```diff
@@ -53,6 +53,15 @@ def GetPlatformName():
 def CountSetBits(n):
     return bin(n).count("1")
 
+def GetEcid():
+    str = os.popen('xxd -p /sys/firmware/acpi/tables/SSDT1 2>/dev/null | tr -d "\\n"').readline()
+    pos1 = str.find('656369643031000e')
+    pos2 = str.find('656369643233000e')
```
Comment (kblaiech) [inline bfsbdump:59]: Question: what are those strings?
656369643031000e
656369643233000e

**Review by kblaiech — COMMENTED** [review]
.

File: bfsbdump (hunk 3 above)
Comment (lsun100) [reply bfsbdump:59]: This is the keyword in ACPI table: string "ecid01" followed by the first 4 bytes and string "ecid23" followed by the 2nd 4 bytes.

Comment (lsun100) [reply bfsbdump:59]: 318           Package () { "ecid01", 0x0102030405060708 },
319           Package () { "ecid23", 0x0BF3000101020304 }

Comment (kblaiech) [reply bfsbdump:59]: Ack

File: bfsbdump (hunk 2 above)
Comment (lsun100) [reply bfsbdump]: done

File: bfsbdump (hunk 1 above)
Comment (lsun100) [reply bfsbdump:67]: done

**Review by kblaiech — APPROVED** [review]

**Pull request merged**